packages/gasket-plugin-intl/lib/index.js

```diff
@@ -1,102 +1,105 @@
-        }
-      };
-    },
-    async create(gasket, { files, pkg }) {
+    configure,
+    async create(gasket, context) {
+      const { files, pkg } = context;
       const rootDir = path.join(__dirname, '..');
       files.add(
-        `${rootDir}/generator/*`,
-        `${rootDir}/generator/**/*`
+        `${ rootDir }/generator/*`,
+        `${ rootDir }/generator/**/*`
       );
 
+      // TODO (@kinetifex): check if react is being added first
       pkg.add('dependencies', {
         '@gasket/intl': devDependencies['@gasket/intl'],
         'react-intl': devDependencies['react-intl']
       });
+
+      context.hasGasketIntl = true;
     },
-    build,
-    middleware(gasket) {
-      //
-      // Make the outputDir available from req for loading locale files
-      // during SSR in HOCs
-      //
-      return (req, res, next) => {
-        req.localesDir = getOutputDir(gasket);
-        next();
+    build: {
+      timing: {
+        first: true
+      },
+      handler: async function build(gasket) {
+        const intlConfig = getIntlConfig(gasket);
+        if (intlConfig.modules) {
+          await buildModules(gasket);
+        }
+        await buildManifest(gasket);
+      }
+    },
+    webpack() {
+      // webpack is not listed as a required dependency but if this lifecycle
+      // is invoked we can assume it has been installed by @gasket/plugin-webpack
+      const webpack = require('webpack');
+
+      return {
+        plugins: [
+          new webpack.EnvironmentPlugin([
+            'GASKET_INTL_LOCALES_DIR',
+            'GASKET_INTL_MANIFEST_FILE'
+          ])
+        ]
       };
     },
-    express(gasket, app) {
-      const outputDir = getOutputDir(gasket);
+    middleware(gasket) {
+      const { defaultLocale, basePath, localesMap } = getIntlConfig(gasket);
 
-      app.use('/_locales', serveStatic(outputDir, {
-        index: false,
-        maxAge: '1y',
-        immutable: true
-      }));
+      return async function intlMiddleware(req, res, next) {
+        /* eslint-disable require-atomic-updates */
+        const acceptLanguage = (req.headers['accept-language'] || defaultLocale).split(',')[0];
+        const locale = await gasket.execWaterfall('intlLocale', acceptLanguage, req, res);
+        const mappedLocale = localesMap && localesMap[locale] || locale;
 
-      //
-      // This is generally loading into Redux during SSR but available otherwise
-      //
-      app.get('/locales-manifest.json', async (req, res) => {
-        const data = loadLocalesManifest(outputDir);
-        res.send(data);
-      });
+        // The gasketData object allows certain config data to be available for
+        // rendering as a global object for access in the browser.
+        res.gasketData = res.gasketData || {};
+
+        // TODO (@kinetifex): This could probably match LocalesProps used by Next.js loaders,
+        //   along with methods on req to preload locale files.
```

Comment:
when? a new major version seems like a good time to make these sorts of changes.

Reply:
Yes, I'm going to do this (and other todos) in a follow-up PR. Just trying to scope back this PR where I could.